ibis/backends/base/sql/alchemy/__init__.py

```diff
@@ -473,6 +474,10 @@ def insert(
                 'yet implemented'
             )
 
+        # If we've been passed a `memtable`, pull out the underlying dataframe
+        if isinstance(in_mem_table := obj.op(), PandasInMemoryTable):
+            obj = in_mem_table.data._df
```

Comment:
```suggestion
        if isinstance(in_mem_table := obj.op(), InMemoryTable):
            obj = in_mem_table.data.to_frame()
```

This will handle any subclass of `InMemoryTable`

Reply:
ahh, cool.  Some other stuff was also borked, but should be fixed up now.

Reply:
Sweet!